modules/serverless-logic-security-oauth-orchestration-in-workflow-example.adoc

```diff
@@ -0,0 +1,302 @@
+        }
+        return new ValidationResult();
+    }
+
+    public static class ValidationResult {
+        private String executionStatus;
+        private String executionStatusMessage;
+        // getters, setters, etc.
+    }
+}
+----
+
+To access the `acme-financial-service` REST service in `currency-exchange-workflow` application, a workflow function such as `getExchangeRate` is used.
+
+.Example of `getExchangeRate` function definition
+[source,json]
+----
+{
+  "name": "getExchangeRate",
+  "type": "rest",
+  "operation": "specs/acme-financial-service.yml#exchangeRate"
+}
+----
+
+In order to filter the information, which must be returned to the currency-exchange-workflow, an `actionDataFilter` is used as follows:
+
+.Example of `actionDataFilter` to pass the `getExchangeRate` results
+[source,json]
+----
+"actionDataFilter": {
+  "results": "${ {exchangeRate: .rate} }"
+}
+----
+
+To calculate the currency exchange rates in `currency-exchange-workflow`, a function named `calculateExchange` is used as follows:
+
+.Example of `calculateExchange` function definition
+[source,json]
+----
+{
+  "name": "calculateExchange",
+  "type": "expression",
+  "operation": "${ { calculateExchangeResult: .amount * .exchangeRate } }"
+}
+----
+
+Similar to `getExchangeRate` to filter the information, which must be returned to the `currency-exchange-workflow`, an `actionDataFilter` is used:
+
+.Example of `actionDataFilter` to pass the `calculateExchange` results
+[source,json]
+----
+ "actionDataFilter": {
+    "results": "${ {result: .calculateExchangeResult} }" 
+  }
+----
+
+The Acme REST service is secured using an OAuth 2.0 `clientCredentials` flow as shown in the following example:
+
+.Example of `acme-financial-service.yml` OpenAPI specification
+[source,yaml]
+----
+openapi: 3.0.3
+info:
+  title: Acme Financial Service API
+  version: 1.0.1
+paths:
+  /financial-service/exchange-rate: <1>
+    get:
+      tags:
+        - Acme Financial Resource
+      operationId: exchangeRate
+      parameters: <2>
+        - name: currencyFrom
+          in: query
+          schema:
+            type: string
+        - name: currencyTo
+          in: query
+          schema:
+            type: string
+        - name: exchangeDate
+          in: query
+          schema:
+            type: string
+      responses: <3>
+        "200":
+          description: OK
+          content: <4>
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExchangeRateResult'
+      security:
+        - acme-financial-oauth: [ ] <5>
+components:
+  schemas:
+    ExchangeRateResult: <6>
+      type: object
+      properties:
+        rate:
+          format: double
+          type: number
+  securitySchemes:
+    acme-financial-oauth: <7>
+      type: oauth2 <8>
+      flows:
+        clientCredentials: <9>
+          authorizationUrl: http://localhost:8281/auth/realms/kogito/protocol/openid-connect/auth
```

Comment:
I think in OpenAPI 3.0, `authorizationUrl` is not used for clientCredentials, only `tokenUrl` is necessary, so it could be deleted for correctness.  
```suggestion
```